### DISCUSSION TITLE
NetBox will not start

### DISCUSSION THREAD

**Discussion by asdettmer** [cisagov/Malcolm#338] [Troubleshooting] [answered]
I consistently get a 502 Bad Gateway when I try to access /NetBox I was careful to select that I wanted Netbox to install during configuration.  I am unsure where the logs are stored for it.

**Comment by mmguero** [comment]
Let's look at a few things:

* the contents of `config/netbox-common.env` in your Malcolm installation directory
* the output of `./scripts logs -s netbox` when run from your Malcolm installation directory (this might be large, you might want to pipe it to a file and share that here)

**Comment by asdettmer** [comment]
Thanks for the assistance getting the logs, looks like I missed setting up the netbox account, I will fix that now

**Comment by mmguero** [comment]
During the [`./scripts/auth_setup`](https://idaholab.github.io/Malcolm/docs/malcolm-hedgehog-e2e-iso-install.html#MalcolmAuthSetup) part of the configuration make sure you select **Y** for `(Re)generate internal passwords for NetBox ?`

**Reply by asdettmer** [reply]
I've rerun auth_setup and selected that option, and then restarted Malcolm, but I'm still getting that error.  Here is the netbox-common.env


>  Parameters related to NetBox (and supporting tools). Note that other more specific parameters
>    can also be configured in the env_file files for netbox* services
>  Whether or not Logstash will enrich network traffic metadata via NetBox API calls
> NETBOX_ENRICHMENT=true
>  The name of the default "site" to be created upon NetBox initialization, and to be used for enrichment
>    if not otherwise specified for the source of the network traffic data
> NETBOX_DEFAULT_SITE=Malcolm
>  Whether or not unobserved network entities in Logstash data will be used to populate NetBox
> NETBOX_AUTO_POPULATE=true
>  Whether or not unobserved network subnets in Logstash data will be created automatically in NetBox
> NETBOX_AUTO_CREATE_PREFIX=true
>  Whether or not services (i.e., destination IP/port) will be looked up during NetBox enrichment
> NETBOX_ENRICHMENT_LOOKUP_SERVICE=true
>  Customize manufacturer matching/creation with NETBOX_AUTO_POPULATE
> NETBOX_DEFAULT_AUTOCREATE_MANUFACTURER=true
> NETBOX_DEFAULT_FUZZY_THRESHOLD=0.95
>  Caching parameters for NetBox's LogStash lookups
> NETBOX_CACHE_SIZE=1000
> NETBOX_CACHE_TTL=30
>  Whether to disable Malcolm's NetBox instance ('true') or not ('false')
> NETBOX_DISABLED=false
> NETBOX_POSTGRES_DISABLED=false
> NETBOX_REDIS_DISABLED=false
> NETBOX_CRON=true


Here is a snippet of the error log (it's just this repeated, but I can share the file if needed)

> netbox-1  | uid=1000(ubuntu) gid=1000(ubuntu) groups=1000(ubuntu),4(adm),5(tty),20(dialout),24(cdrom),25(floppy),27(sudo),29(audio),30(dip),44(video),46(plugdev)
> netbox-1  | django.db.utils.OperationalError: connection failed: connection to server at "172.29.1.17", port 5432 failed: FATAL:  password authentication failed for user "netbox"
> netbox-1  | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
> netbox-1  | ⏳ Waiting on DB... (0s / 30s)
> netbox-1  | django.db.utils.OperationalError: connection failed: connection to server at "172.29.1.17", port 5432 failed: FATAL:  password authentication failed for user "netbox"
> netbox-1  | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
> netbox-1  | ⏳ Waiting on DB... (3s / 30s)
> netbox-1  | django.db.utils.OperationalError: connection failed: connection to server at "172.29.1.17", port 5432 failed: FATAL:  password authentication failed for user "netbox"
> netbox-1  | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
> netbox-1  | ⏳ Waiting on DB... (6s / 30s)
> netbox-1  | django.db.utils.OperationalError: connection failed: connection to server at "172.29.1.17", port 5432 failed: FATAL:  password authentication failed for user "netbox"
> netbox-1  | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
> netbox-1  | ⏳ Waiting on DB... (9s / 30s)
> netbox-1  | django.db.utils.OperationalError: connection failed: connection to server at "172.29.1.17", port 5432 failed: FATAL:  password authentication failed for user "netbox"
> netbox-1  | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
> netbox-1  | ⏳ Waiting on DB... (12s / 30s)
> netbox-1  | django.db.utils.OperationalError: connection failed: connection to server at "172.29.1.17", port 5432 failed: FATAL:  password authentication failed for user "netbox"
> netbox-1  | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
> netbox-1  | ⏳ Waiting on DB... (15s / 30s)
> netbox-1  | django.db.utils.OperationalError: connection failed: connection to server at "172.29.1.17", port 5432 failed: FATAL:  password authentication failed for user "netbox"
> netbox-1  | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
> netbox-1  | ⏳ Waiting on DB... (18s / 30s)
> netbox-1  | django.db.utils.OperationalError: connection failed: connection to server at "172.29.1.17", port 5432 failed: FATAL:  password authentication failed for user "netbox"
> netbox-1  | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
> netbox-1  | ⏳ Waiting on DB... (21s / 30s)
> netbox-1  | django.db.utils.OperationalError: connection failed: connection to server at "172.29.1.17", port 5432 failed: FATAL:  password authentication failed for user "netbox"
> netbox-1  | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
> netbox-1  | ⏳ Waiting on DB... (24s / 30s)
> netbox-1  | django.db.utils.OperationalError: connection failed: connection to server at "172.29.1.17", port 5432 failed: FATAL:  password authentication failed for user "netbox"
> netbox-1  | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
> netbox-1  | ⏳ Waiting on DB... (27s / 30s)
> netbox-1  | ❌ Waited 30s or more for the DB to become ready.
> netbox-1 exited with code 0

**Comment by mmguero** [comment, marked as answer]
Hmmm... the stuff that's failing there should all be internal and not tied to anything you're doing. I wonder if we need to clear out the netbox database for some reason and start fresh. If you're okay doing a `./scripts/wipe` which will clear all of Malcolm's data, that should do it, or if not do a `./scripts/stop` followed by `rm -rf ./netbox/postgres/*`, then try restarting and see what happens.

**Reply by asdettmer** [reply]
success! Since this is just a lab test for now, there is no issue clearing data.  I ran the wipe script, then start, and Netbox is up and running.  Thank you!!

**Reply by mmguero** [reply]
Great! No problem.
